### PR TITLE
[AIE2] Offload PADD to slot S too

### PR DESCRIPTION
**Pull request by krishnamtibrewala**

### CODE REVIEW

File: llvm/test/CodeGen/AIE/aie2/schedule/pseudo-slotinfo.mir [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 #
 # (c) Copyright 2023-2024 Advanced Micro Devices, Inc. or its affiliates
-# RUN: llc -march=aie2 -run-pass=postmisched %topdown-multi %s -o - | FileCheck %s
+# RUN: llc -march=aie2 -run-pass=postmisched --aie-bottomup-cycles=0 %s -o - | FileCheck %s
```
Comment (krishnamtibrewala) [inline llvm/test/CodeGen/AIE/aie2/schedule/pseudo-slotinfo.mir]: FIXME

Comment (gbossu) [reply llvm/test/CodeGen/AIE/aie2/schedule/pseudo-slotinfo.mir]: With #91 merged, you can now use the `-s` option when calling `update_xxxx_test_checks.py`

File: llvm/test/CodeGen/AIE/aie2/GlobalISel/inst-select-no-combine.mir [hunk 2]
```diff
@@ -74,8 +74,8 @@ body:             |
     ; CHECK-NEXT: {{  $}}
     ; CHECK-NEXT: [[COPY:%[0-9]+]]:vec256 = COPY $wl0
     ; CHECK-NEXT: [[COPY1:%[0-9]+]]:ep = COPY $p0
-    ; CHECK-NEXT: [[PADD_imm_pseudo:%[0-9]+]]:ep = PADD_imm_pseudo [[COPY1]], 992
-    ; CHECK-NEXT: VST_dmw_sts_w_ag_idx_imm [[COPY]], [[PADD_imm_pseudo]], 0 :: (store (<8 x s32>))
```
Comment (martien-de-jong) [inline llvm/test/CodeGen/AIE/aie2/GlobalISel/inst-select-no-combine.mir:78]: Would it reduce the diff noise if we kept PADD_imm_pseudo for the most versatile ABS case? We would just have to tighten its constraints, and then define the appropriate A, B, S, AB, AS, BS intersections

File: llvm/lib/Target/AIE/AIE2MultiSlotPseudoInstrInfo.td [hunk 3]
```diff
@@ -19,11 +19,11 @@
 let mayLoad = false, mayStore = false, hasSideEffects = false, Itinerary = II_PADD in {
   let Constraints = "$ptr = $res" in {
     def PADD_mod_pseudo : MultiSlot_Pseudo< (outs eP:$res), (ins eP:$ptr, eM:$mod),
-                          "padd_mod_pseudo", "[$ptr], $mod", 
-                          [PADDB_ldb_ptr_inc_nospill_nrm, PADDA_lda_ptr_inc_idx] >;
+                          "padd_mod_pseudo", "[$ptr], $mod", [PADDB_ldb_ptr_inc_nospill_nrm, PADDS_st_ptr_inc_idx, PADDA_lda_ptr_inc_idx] >;
     def PADD_imm_pseudo : MultiSlot_Pseudo< (outs eP:$res), (ins eP:$ptr, imm9x4:$imm),
```
Comment (gbossu) [inline llvm/lib/Target/AIE/AIE2MultiSlotPseudoInstrInfo.td]: To be consistent with the naming of multi-slot loads, maybe change to PADD_imm10_pseudo and PADD_imm9_pseudo?

**Review by gbossu — APPROVED** [review]
LGTM

**Pull request merged**